pages/en/zkapps/how-to-test-a-zkapp.mdx

```diff
@@ -37,8 +37,97 @@ Because we are using Jest, it's helpful to break down all functionality of your
 
 For examples of existing tests, we recommend creating a template example using the Mina zkApp CLI via `zk project <name>` and examining the test file there. [You will see a basic example of a few tests](https://github.com/o1-labs/zkapp-cli/blob/main/templates/project-ts/src/Add.test.ts) that deploy and update the state on a smart contract using Jest.
 
+#### Creating a local blockchain
+
+The `Mina.LocalBlockchain()` method specifies a mock Mina ledger of accounts that runs locally. It contains logic for updating the ledger that can be used to test your smart contract locally.
+
+```ts
+let Local = Mina.LocalBlockchain();
+Mina.setActiveInstance(Local);
+```
+
+#### Deploying a contract locally
+
+The mock Mina local blockchain contains test accounts that can be used to deploy the contract and pay transaction fees.
+
+```ts
+// test accounts that pays all the fees, and puts additional funds into the zkapp.
+let feePayer = Local.testAccounts[0].privateKey;
+```
+
+We generate a zkApp account and a new instance of the smart contract to deploy locally for testing.
+
+```ts
+// zkapp account
+let zkAppPrivateKey = PrivateKey.random();
+let zkAppAddress = zkAppPrivateKey.toPublicKey();
+let zkAppInstance = new Add(zkAppAddress);
+```
+
+Use the test account and zkApp keys to construct a transaction to pay account creation fees and deploy your smart contract. This transaction is sent to the local blockchain. 
+
+```ts
+let txn = await Mina.transaction(feePayer, () => {
+  Party.fundNewAccount(feePayer);
+  zkAppInstance.deploy({ zkappKey: zkAppPrivateKey});
+});
+txn.send();
+```
+
+#### Writing integration tests
+
+While Jest can be used to write both unit and integration tests, It is reccomended to create linear scripts for smart contract integration tests. The scripting worflow is easier to reason about when creating large interconnected tests. It also makes debugging easier and provides good error tracing. 
```

Comment:
Not sure I agree on this. I believe it's a better convention to teach people how to use Jest, otherwise their code could break and they won't know until they decide to run the script again and hopefully notice it

Reply:
We decided to use script files for the vm integration tests after the discussion on this [PR](https://github.com/o1-labs/snarkyjs/pull/295). The scripts are added to run in CI. We can recommend as convention to add the integration tests to Jest if you prefer.

Reply:
For the vm integration tests, anything is fine as long because it's hooked into CI.

But for this, our recommended convention is important b/c affects the dev experience and opens the door for more bugs in the zkApp ecosystem if we're not hooking tests into the test runner we've chosen.

In response to Gregor's comment on the linked issue, if a failed test needs 1,000+ lines for error output (!) and Jest can't show it, then we should create a separate issue to look into that. If you agree with him, could you open an issue for that on the CLI please, with a mention of how to reproduce or we keep an eye out to add that if it ever occurs, please?

Reply:
I agree we have to be careful about what recommendations and conventions we set for the community. I'm fine to include Jest in our docs as convention. I'll rewrite this portion with Jest as the recommendation and then create a seperate issue to address error output.

Reply:
Thanks!